Libraries/src/Amazon.Lambda.RuntimeSupport/Helpers/Logging/MessageProperty.cs

```diff
@@ -0,0 +1,171 @@
+            }
+
+            var idxOfDelimeter = messageToken.IndexOfAny(PARAM_FORMAT_DELIMITERS);
+            if (idxOfDelimeter < 0)
+            {
+                this.Name = messageToken.ToString().Trim();
+                this.FormatArgument = null;
+            }
+            else
+            {
+                this.Name = messageToken.Slice(0, idxOfDelimeter).ToString().Trim();
+                this.FormatArgument = messageToken.Slice(idxOfDelimeter + 1).ToString().Trim();
+                if(this.FormatArgument == string.Empty)
+                {
+                    this.FormatArgument = null;
+                }
+            }
+        }
+
+        /// <summary>
+        /// The original text of the message property.
+        /// </summary>
+        public string MessageToken { get; private set; }
+
+        public enum Directive { Default, JsonSerialization };
+
+        /// <summary>
+        /// The Name of the message property.
+        /// </summary>
+        public string Name { get; private set; }
+
+        /// <summary>
+        /// Optional format argument. If used the value will be formatted using string.Format passing in this format argument.
+        /// </summary>
+        public string FormatArgument { get; private set; }
+
+        /// <summary>
+        /// Optional format directive. Gives users the ability
+        /// to indicate when types should be serialized to JSON when using structured logging.
+        /// </summary>
+        public Directive FormatDirective { get; private set; }
+
+        /// <summary>
+        /// Formats the value as a string that can be used to replace the message property token inside a message template.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public string FormatForMessage(object value)
+        {
+            if(value is byte[] bytes)
+            {
+                return FormatByteArray(bytes);
+            }
+            if (value is ReadOnlyMemory<byte> roBytes)
+            {
+                return FormatByteArray(roBytes.Span);
+            }
+            if (value is Memory<byte> meBytes)
+            {
+                return FormatByteArray(meBytes.Span);
+            }
+            if (value == null || value is IList || value is IDictionary)
+            {
+                return this.MessageToken;
+            }
+            if(!string.IsNullOrEmpty(this.FormatArgument))
+            {
+                return ApplyFormatArgument(value, this.FormatArgument);
+            }
+            if(value is DateTime dt)
+            {
+                return dt.ToString(AbstractLogMessageFormatter.DateFormat, CultureInfo.InvariantCulture);
+            }
+            if (value is DateTimeOffset dto)
+            {
+                return dto.ToString(AbstractLogMessageFormatter.DateFormat, CultureInfo.InvariantCulture);
+            }
+
+            return value.ToString();
```

Comment:
Throughout, could/should we support `DateOnly`/`TimeOnly`?

Reply:
Done
```
var dateOnly = new DateOnly(2024, 2, 18);
var timeOnly = new TimeOnly(15, 19, 45, 545);
context.Logger.LogInformation("DateOnly: {dateOnly}, TimeOnly: {timeOnly}", dateOnly, timeOnly);
```

```json
{
  "timestamp": "2024-04-02T00:11:35.110Z",
  "level": "Information",
  "requestId": "5aa85d4a-1986-4b36-8b50-1adf29d96f0b",
  "traceId": "Root=1-660b4d36-22f221a17181ce5b3cef6f4e;Parent=0124753359b8279a;Sampled=0;Lineage=cd25c10f:0",
  "message": "DateOnly: 2024-02-18, TimeOnly: 15:19:45.545",
  "dateOnly": "2024-02-18",
  "timeOnly": "15:19:45.545"
}
```